nibabel/brainvoyager/bv.py

```diff
@@ -0,0 +1,952 @@
+    parent_hdr_dict: OrderedDict
+       When parse_BV_header() is called recursively the already filled
+       (parent) hdr_dict is passed to give access to n_fields_name fields
+       outside the current scope (see below).
+
+    Returns
+    -------
+    hdr_size : int
+        Size of header when packed into bytes ready for writing to file.
+    """
+    hdr_size = 0
+    for name, format, def_or_name in hdr_dict_proto:
+        value = hdr_dict[name]
+        # handle zero-terminated strings
+        if format == 'z':
+            hdr_size += len(value) + 1
+        # handle array fields
+        elif isinstance(format, tuple):
+            # check the length of the array to expect
+            if def_or_name in hdr_dict:
+                n_values = hdr_dict[def_or_name]
+            else:
+                n_values = parent_hdr_dict[def_or_name]
+            for i in range(n_values):
+                # recursively iterate through the fields of all items
+                # in the array
+                hdr_size += calc_BV_header_size(format, value[i], hdr_dict)
+        # handle conditional fields
+        elif isinstance(def_or_name, tuple):
+            if hdr_dict[def_or_name[1]] == def_or_name[2]:
+                hdr_size += calcsize(format)
+            else:
+                continue
+        else:
+            hdr_size += calcsize(format)
+    return hdr_size
+
+
+def update_BV_header(hdr_dict_proto, hdr_dict_old, hdr_dict_new,
+                     parent_old=None, parent_new=None):
+    """Update a hdr_dict after changed nested-loops-number or conditional fields.
+
+    This function can be (and is) called recursively to iterate through nested
+    fields (e.g. the prts field of the VTC header).
+
+    Parameters
+    ----------
+    hdr_dict_proto: tuple
+        tuple of format described in Notes of :func:`parse_BV_header`
+    hdr_dict_old: OrderedDict
+       hdr_dict before any changes.
+    hdr_dict_new: OrderedDict
+       hdr_dict with changed fields in n_fields_name or c_fields_name fields.
+    parent_old: OrderedDict
+       When update_BV_header() is called recursively the not yet updated
+       (parent) hdr_dict is passed to give access to n_fields_name fields
+       outside the current scope (see below).
+    parent_new: OrderedDict
+       When update_BV_header() is called recursively the not yet updated
+       (parent) hdr_dict is passed to give access to n_fields_name fields
+       outside the current scope (see below).
+
+    Returns
+    -------
+    hdr_dict_new : OrderedDict
+        An updated version hdr_dict correcting effects of changed nested and
+        conditional fields.
+    """
+    for name, format, def_or_name in hdr_dict_proto:
+        # handle nested loop fields
+        if isinstance(format, tuple):
+            # calculate the change of array length and the new array length
+            if def_or_name in hdr_dict_old:
+                delta_values = hdr_dict_new[def_or_name] - \
```

Comment:
Vague preference for parentheses round expression to allow line breaks, rather than continuation character `\` : https://www.python.org/dev/peps/pep-0008/#id19